### PR TITLE
Feature/display state manager

### PR DESCRIPTION
**Pull request by Mn0491**
This pull request will implement a persistent clock that is displayed on the MarkI faceplate. It also has a display state manager which allows mycroft core the ability to know which skill currently using the markI faceplate

A few dependencies are needed for this to work

mycroft-core/requirements.txt 
pillow==4.1.1

sudo apt-get install python-dev
sudo apt-get install libjpeg-dev

**skill-date-time also needs to be merged from dev into masters**

**MycroftAI/enclosure - need to merge display-update and install-script-fix into the master branch of enclosure and upload code into the arduino**

### CODE REVIEW

File: mycroft/client/enclosure/mouth.py [hunk 1]
```diff
@@ -70,3 +74,35 @@ def text(self, event=None):
...
+        xOffset = ""
+        yOffset = ""
+        clearPrevious = ""
+        if event and event.data:
+            code = event.data.get("img_code", code)
+            xOffset = event.data.get("xOffset", xOffset)
+            yOffset = event.data.get("yOffset", yOffset)
+            clearPrevious = event.data.get("clearPrev", clearPrevious)
+
+        # clearAfter = int(cleartAfter) # convert clearAfter to 0 or 1
```
Comment (MatthewScholefield) [inline mycroft/client/enclosure/mouth.py]: Extra code commented

File: mycroft/client/enclosure/display_manager.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+def remove_active():
+    """ Remove the active skill in the skill manager
+    """
+    LOG.error("Removing active skill...")
+    write_data({"active_skill": ""})
+
+
+def run():
+    """ TODO: document
```
Comment (MatthewScholefield) [inline mycroft/client/enclosure/display_manager.py]: Perhaps rename to `initiate_ws` or `initiate_display_manager_ws` (like it was imported as) and make the doc string the comment written near the function ("initiates the web sockets on display manager").

**Comment by Mn0491** [comment]
Changes made

**Comment by augustnmonteiro** [comment]
I think that would be better change the `initiate_display_manager_ws` to a class

**Comment by ethanaward** [comment]
There's also a git conflict that needs to be resolved.

**Comment by penrods** [comment]
Let's merge this next release (the one after July 6th) along with the changes to Enclosure, etc.

**Pull request merged**